lib/qiita/events/advent_calendar_2022_kansousyou.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Qiita.Events.AdventCalendar2022Kansousyou do
+  alias Qiita.Events.AdventCalendar2022Kansousyou.MarkdownGenerator
+  alias Qiita.Events.Repo
+
+  def run do
+    tags = ["Elixir", "40代駆け出しエンジニア", "AdventCalendar2022", "猪木", "闘魂"]
+    title = "闘魂Elixir ── 03. 完走賞を目指してみましょう！【自動更新記事】"
+    id = "17d55cf896c24b13350e"
+
+    "https://qiita.com/advent-calendar/2022/elixir"
+    |> Req.get!()
+    |> Map.get(:body)
+    |> String.split("encryptedId")
+    |> Enum.map(fn s -> Regex.named_captures(~r/"user":{"urlName":"(?<user>.*?)",/, s) end)
+    |> Enum.slice(2..-1//1)
+    |> Enum.map(fn %{"user" => user} -> user end)
+    |> Enum.frequencies()
+    |> Enum.sort_by(fn {_user, cnt} -> cnt end, :desc)
+    |> MarkdownGenerator.generate()
+    |> Repo.post_markdown(tags, title, id)
+  end
```

Comment:
Seems like a large part of the code here is for parsing response body so body parser can be separated as a function. 

Below is an example:

```suggestion
    Req.get("https://qiita.com/advent-calendar/2022/elixir")
    |> parse_qiita_response(body)
    |> MarkdownGenerator.generate()
    |> Repo.post_markdown(tags, title, id)
  end

  defp parse_qiita_response(%{status: 200, body: body}) do
    extract_users(body)
    |> Enum.frequencies_by(fn %{"user" => user} -> user end)
    |> Enum.sort_by(fn {_user, cnt} -> cnt end, :desc)
  end

  defp parse_qiita_response(%{status: _, body: _}) do
    raise("Could not fetch data from Qiita")
  end

  defp extract_users(body) when is_binary(body) do
    body
    |> String.split("encryptedId")
    |> Enum.map(fn s -> Regex.named_captures(~r/"user":{"urlName":"(?<user>.*?)",/, s) end)
    |> Enum.reject(&is_nil/1)
  end
```

Disclaimer: This is an imaginary code in my head and it is not tested.

Reply:
Thanks !

I use https://elixirjp.slack.com/archives/C01JMMZM0SH/p1669489613405639?thread_ts=1669479293.683269&cid=C01JMMZM0SH .
I want to use floki!
This code is LGTM.

Thank you very much.